libcodechecker/analyze/analyzers/analyzer_clang_tidy.py

```diff
@@ -14,12 +14,16 @@
 import shlex
 import subprocess
 
+from libcodechecker.analyze import analyzer_env
+from libcodechecker.analyze import host_check
 from libcodechecker.analyze.analyzers import analyzer_base
-from libcodechecker.logger import get_logger
-from libcodechecker.util import get_binary_in_path
-from libcodechecker.analyze.analyzer_env import\
-    extend_analyzer_cmd_with_resource_dir
+from libcodechecker.analyze.analyzers import config_handler_clang_tidy
 from libcodechecker.analyze.analyzers import result_handler_clang_tidy
+from libcodechecker.analyze.analyzers.analyzer_clangsa import ClangSA
```

Comment:
Shouldn't we create a separate binary detection part for each analyzers so the tidy analyzer will not depend on the `analyzer_clangsa` modul?

Reply:
We definitely should. It is a workaround below in this file, that we cannot get the resource dir from clang-tidy at the moment. As soon as we come up with a way to get the resource dir without looking at the clang binary, we can get rid of this dependency. There is a fixme for that in this file.